test/functional/tests.py

```diff
@@ -807,6 +807,68 @@ def test_bitswap_unwant(self):
         result = self.api.bitswap_unwant(key='QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V')
         self.assertTrue(result is not None)
 
+@skipIfOffline()
+class IpfsApiPubSubTest(unittest.TestCase):
+
+    def setUp(self):
+        self.api = ipfsapi.Client()
+
+    def test_pubsub_pub_sub(self):
+        """
+        We should test both publishing and subscribing
+        at the same time for it's difficult
+        to determine if one works without checking the
+        other.
+        """
+        # the topic that will be published/subscribed to
+        topic = 'testing'
+        # the message that will be published
+        message = 'hello'
+
+        expected_data = 'aGVsbG8='	
+        expected_topicIDs = [topic]
+
+
+        # get the subscription stream
+        sub = self.api.pubsub_sub(topic)
+
+        # make sure something was actually returned from the subscription
+        assert sub is not None
+
+        # publish a message to topic
+        self.api.pubsub_pub(topic, message)
+
+        # get the 
+        sub_data = next(sub)
+
+        self.assertEqual(sub_data['data'], expected_data)
+        self.assertEqual(sub_data['topicIDs'], expected_topicIDs)
+
+    def test_pubsub_ls(self):
+        """
+        Testing the ls, assumes we are able
+        to at least subscribe to a topic
+        """
+        topic = 'lsing'
+        expected_return = { 'Strings': [topic] }
+
+        # subscribe to the topic testing
+        # in this test we don't care about the return value
```

Comment:
Maybe add something like “… but we need to keep it around for the lifetime of this function to ensure that we stay subscribed to the topic long enough” to make it more obvious what is happening?

Reply:
This reminds me, when requests performs a request to the subscription url, I believe it will remain open. This could waste resources if someone is constantly subscribing and discarding channels. I don't think it'll close itself because it's still technically listening to the channel. What do you recommend doing for this? Wrapping it in a class that has a function which will close the subscription?